CONTRIBUTING.md

```diff
@@ -42,8 +44,90 @@ mvn -f pom.client.xml -Dgpg.skip -DskipTests clean install
+The dependency-version should be set to the most recent released version and the current-version is set to the next version to be released.
+
+For example:
+
+`com.azure:azure-identity;1.0.0-preview.4;1.0.0-preview.5`
+
+Note: In the case of a new or unreleased artifact both versions will be the same.
+
+For more information 
+
+### Libraries vs External Dependencies
+
+Libraries refer to things that are built and released as part of the Azure SDK. Libraries have a current version and a dependency version. 
+
+External Dependencies refer to dependencies for things that are not built and released as part of the Azure SDK regardless of the source. External Dependencies will only ever have one version.
+
+### Current version vs Depdendency version
+
+Current version - This is the version we should be using when defining a component in its POM file and also when dependent components are built within the same pipeline. The current version is the version currently in development.
+Depdendency version - This is the version we should be using when a given library is a dependency outside of a particular area. This should be the latest released version of the package.
+
+For example: `com.azure:azure-storage-blob-batch` has dependencies on `com.azure:azure-core`, `com.azure:azure-core-http-netty` and `com.azure:azure-storage-blob`. Because `com.azure:azure-core` and `com.azure:azure-core-http-netty` are both built outside of azure-storage pipeline we should be using the released or *Dependency* versions of these when they're dependencies of another library. Similarly, libraries built as part of the same pipeline, that have interdependencies, should be using the Current version. Since `com.azure:azure-storage-blob-batch` and `com.azure:azure-storage-blob` are both built part of the azure-batch pipeline when `com.azure:azure-storage-blob` is declared as a dependency of `com.azure:azure-storage-blob-batch` it should be the *Current* version.
+
+This is going to be especially important after GA when releases aren't going to be the entire Azure SDK every time. If we're releasing a patch for a targeted azure-storage fix then we shouldn't need to build and release azure-core, we should be targeting the released versions and only building/releasing that update to azure-storage. It's worth noting that right now, in the version_client.txt, the dependency/current versions are the same. This will change once we GA, at which point the current version should be ahead of the dependency version.
+
+What about README files? Right now the README files in the repro end up getting into an odd state since things like samples and versions can get updated during the development process. We're in the process of versioning documentation with the releases which means that the docs are snapshot at the time of the release and then versioned and stored. This will allow the README files in the repro to have updated samples and versions that are setup for the next release.
```

Comment:
"repro" -> "repo"

Reply:
Will be fixed in the next update of https://github.com/Azure/azure-sdk-for-java/pull/5889